l10n_fr_hr_ssnid_validation/models/hr_employee.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from stdnum import get_cc_module
+from stdnum.exceptions import (
+    InvalidChecksum,
+    InvalidComponent,
+    InvalidFormat,
+    InvalidLength,
+)
+
+class HrEmployee(models.Model):
+    _inherit = 'hr.employee'
+
+    ssnid_key = fields.Char(
+        string='SSN No Key',
+        help='Social Security Number Key',
+        size=2,
+        required=True,
+    )
+
+    @api.onchange('ssnid', 'ssnid_key')
+    def onchange_ssnid(self):
+        self.check_ssnid()
```

Comment:
isn't constrain not enough?

Reply:
The constraint apply when pushing the Save button/write call I think, and it could help the user if he is warned directly when changing the value on the form?

Reply:
as it is two fields it will be called when one field empty and of course will fail the check, that is annoying